Source/WebCore/rendering/RenderQuote.cpp

```diff
@@ -203,160 +211,161 @@ static int quoteTableLanguageComparisonFunction(const void* a, const void* b)
-        { "rn"_s,         0, 0x201d, 0x201d, 0x2019, 0x2019 },
-        { "ro"_s,         0, 0x201e, 0x201d, 0x00ab, 0x00bb },
-        { "rof"_s,        0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "ru"_s,         0, 0x00ab, 0x00bb, 0x201e, 0x201c },
-        { "rw"_s,         0, 0x00ab, 0x00bb, 0x2018, 0x2019 },
-        { "rwk"_s,        0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "saq"_s,        0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "sbp"_s,        0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "seh"_s,        0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "ses"_s,        0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "sg"_s,         0, 0x00ab, 0x00bb, 0x201c, 0x201d },
-        { "shi"_s,        1, 0x00ab, 0x00bb, 0x201e, 0x201d },
-        { "shi-tfng"_s,   0, 0x00ab, 0x00bb, 0x201e, 0x201d },
-        { "si"_s,         0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "sk"_s,         0, 0x201e, 0x201c, 0x201a, 0x2018 },
-        { "sl"_s,         0, 0x201e, 0x201c, 0x201a, 0x2018 },
-        { "sn"_s,         0, 0x201d, 0x201d, 0x2019, 0x2019 },
-        { "so"_s,         0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "sq"_s,         0, 0x201e, 0x201c, 0x201a, 0x2018 },
-        { "sr"_s,         1, 0x201e, 0x201c, 0x201a, 0x2018 },
-        { "sr-latn"_s,    0, 0x201e, 0x201c, 0x201a, 0x2018 },
-        { "sv"_s,         0, 0x201d, 0x201d, 0x2019, 0x2019 },
-        { "sw"_s,         0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "swc"_s,        0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "ta"_s,         0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "te"_s,         0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "teo"_s,        0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "th"_s,         0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "ti-er"_s,      0, 0x2018, 0x2019, 0x201c, 0x201d },
-        { "to"_s,         0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "tr"_s,         0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "twq"_s,        0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "tzm"_s,        0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "uk"_s,         0, 0x00ab, 0x00bb, 0x201e, 0x201c },
-        { "ur"_s,         0, 0x201d, 0x201c, 0x2019, 0x2018 },
-        { "vai"_s,        1, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "vai-latn"_s,   0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "vi"_s,         0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "vun"_s,        0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "xh"_s,         0, 0x2018, 0x2019, 0x201c, 0x201d },
-        { "xog"_s,        0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "yav"_s,        0, 0x00ab, 0x00bb, 0x00ab, 0x00bb },
-        { "yo"_s,         0, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "zh"_s,         1, 0x201c, 0x201d, 0x2018, 0x2019 },
-        { "zh-hant"_s,    0, 0x300c, 0x300d, 0x300e, 0x300f },
-        { "zu"_s,         0, 0x201c, 0x201d, 0x2018, 0x2019 },
+    static const std::array quoteTable {
```

Comment:
```suggestion
    static constexpr std::array quoteTable {
```

Reply:
This doesn't build at the moment:
```
In file included from /Volumes/Work/WebKit/OpenSource/WebKitBuild/Debug/DerivedSources/WebCore/unified-sources/UnifiedSource397.cpp:6:
/Volumes/Work/WebKit/OpenSource/Source/WebCore/rendering/RenderQuote.cpp:213:33: error: constexpr variable 'quoteTable' must be initialized by a constant expression
  213 |     static constexpr std::array quoteTable {
  214 |         QuotesForLanguage { "af"_span,         0, 0x201c, 0x201d, 0x2018, 0x2019 },
  215 |         QuotesForLanguage { "agq"_span,        0, 0x201e, 0x201d, 0x201a, 0x2019 },
  216 |         QuotesForLanguage { "ak"_span,         0, 0x201c, 0x201d, 0x2018, 0x2019 },
  217 |         QuotesForLanguage { "am"_span,         0, 0x00ab, 0x00bb, 0x2039, 0x203a },
  218 |         QuotesForLanguage { "ar"_span,         0, 0x201d, 0x201c, 0x2019, 0x2018 },
  219 |         QuotesForLanguage { "asa"_span,        0, 0x201c, 0x201d, 0x2018, 0x2019 },
  220 |         QuotesForLanguage { "az-cyrl"_span,    0, 0x00ab, 0x00bb, 0x2039, 0x203a },
  221 |         QuotesForLanguage { "bas"_span,        0, 0x00ab, 0x00bb, 0x201e, 0x201c },
  222 |         QuotesForLanguage { "bem"_span,        0, 0x201c, 0x201d, 0x2018, 0x2019 },
  223 |         QuotesForLanguage { "bez"_span,        0, 0x201c, 0x201d, 0x2018, 0x2019 },
  224 |         QuotesForLanguage { "bg"_span,         0, 0x201e, 0x201c, 0x201a, 0x2018 },
  225 |         QuotesForLanguage { "bm"_span,         0, 0x00ab, 0x00bb, 0x201c, 0x201d },
  226 |         QuotesForLanguage { "bn"_span,         0, 0x201c, 0x201d, 0x2018, 0x2019 },
  227 |         QuotesForLanguage { "br"_span,         0, 0x00ab, 0x00bb, 0x2039, 0x203a },
  228 |         QuotesForLanguage { "brx"_span,        0, 0x201c, 0x201d, 0x2018, 0x2019 },
In file included from /Volumes/Work/WebKit/OpenSource/WebKitBuild/Debug/DerivedSources/WebCore/unified-sources/UnifiedSource397.cpp:1:
In file included from /Volumes/Work/WebKit/OpenSource/Source/WebCore/WebCorePrefix.h:164:
In file included from /Volumes/Work/WebKit/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/FastMalloc.h:26:
/Volumes/Work/WebKit/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/StdLibExtras.h:782:25: note: reinterpret_cast is not allowed in a constant expression
  782 |     return ReturnType { reinterpret_cast<T*>(const_cast<std::remove_const_t<U>*>(span.data())), span.size_bytes() / sizeof(T) };
/Volumes/Work/WebKit/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/StdLibExtras.h:917:95: note: in call to 'spanReinterpretCast({&"af"[0], 2})'
  917 |     template<ByteType U> static constexpr auto cast(std::span<const T, Extent> span) { return spanReinterpretCast<const U>(span); }
/Volumes/Work/WebKit/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/StdLibExtras.h:922:12: note: in call to 'cast({&"af"[0], 2})'
```

Reply:
```
constexpr std::span<const LChar> operator""_span(const char* characters, size_t n)
{
    auto span = byteCast<LChar>(unsafeMakeSpan(characters, n));
```

`byteCast<LChar>()` ends up calling `spanReinterpretCast<>()` which calls `reinterpret_cast<>()` which is not not allowed with constexpr.

Reply:
I'll try and fix this is in a follow-up. It should be able to make this truly `constexpr`.

Reply:
Actually, std::bit_cast between pointer types is not constexpr so it doesn't seem trivial to fix.